velox/functions/prestosql/StringFunctions.cpp

```diff
@@ -143,6 +143,50 @@ class UpperLowerTemplateFunction : public exec::VectorFunction {
  * */
 class ConcatFunction : public exec::VectorFunction {
  public:
+  ConcatFunction(
+      const std::string& name,
+      const std::vector<exec::VectorFunctionArg>& inputArgs) {
+    auto numArgs = inputArgs.size();
+
+    // Save constant values to constantStrings_ and constantStringViews_.
+    // Identify and combine consecutive constant inputs.
+    argMapping_.reserve(numArgs);
+    constantStrings_.reserve(numArgs);
+    constantStringViews_.reserve(numArgs);
+
+    for (auto i = 0; i < numArgs; ++i) {
+      argMapping_.push_back(i);
+
+      const auto& arg = inputArgs[i];
+      if (arg.constantValue) {
+        std::string value = arg.constantValue->as<ConstantVector<StringView>>()
+                                ->valueAt(0)
+                                .str();
+
+        column_index_t j = i + 1;
+        for (; j < inputArgs.size(); ++j) {
+          if (!inputArgs[j].constantValue) {
+            break;
+          }
+
+          value += inputArgs[j]
+                       .constantValue->as<ConstantVector<StringView>>()
+                       ->valueAt(0)
+                       .str();
+        }
+
+        constantStrings_.push_back(std::string(value.data(), value.size()));
```

Comment:
nit: can you make a comment here since we have reserved the space for constantStrings_ so it is safe to take string view here while building constantStrings_ vector? Thanks!

Reply:
Good point. I moved this into a separate loop as we discussed offline.